scripts/linux/cross-platform-rust-build.sh

```diff
@@ -0,0 +1,315 @@
+# Obtain and validate the options supported by this script
+###############################################################################
+process_args()
+{
+    save_next_arg=0
+    for arg in "$@"
+    do
+        if [ $save_next_arg -eq 1 ]; then
+            PACKAGE_OS="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 2 ]; then
+            PACKAGE_ARCH="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 3 ]; then
+            BUILD_PATH="$arg"
+            save_next_arg=0
+        else
+            case "$arg" in
+                "--os" ) save_next_arg=1;;
+                "--arch" ) save_next_arg=2;;
+                "--build-path" ) save_next_arg=3;;
+                "-h" | "--help" ) usage;;
+                * ) usage;;
+            esac
+        fi
+    done
+}
+
+process_args "$@"
+
+[[ -z "$PACKAGE_OS" ]] && { print_error 'OS is a required parameter'; exit 1; }
+[[ -z "$PACKAGE_ARCH" ]] && { print_error 'Arch is a required parameter'; exit 1; }
+[[ -z "$BUILD_PATH" ]] && { print_error 'Build path is a required parameter'; exit 1; }
+
+set -e
+
+# Get directory of running script
+DIR="$(cd "$(dirname "$0")" && pwd)"
+
+BUILD_REPOSITORY_LOCALPATH="$(realpath "${BUILD_REPOSITORY_LOCALPATH:-$DIR/../..}")"
+
+REVISION="${REVISION:-1}"
+
+CMAKE_ARGS='-DCMAKE_BUILD_TYPE=Release'
+CMAKE_ARGS="$CMAKE_ARGS -DBUILD_SHARED=On -Drun_unittests=Off -Duse_default_uuid=On -Duse_emulator=Off -Duse_http=Off"
+
+DOCKER_VOLUME_MOUNTS=''
+
+case "$PACKAGE_OS" in
+    'alpine')
+        DOCKER_IMAGE='ubuntu:18.04'
+        ;;
+
+    'ubuntu18.04')
+        DOCKER_IMAGE='ubuntu:18.04'
+        ;;
+esac
+
+if [ -z "$DOCKER_IMAGE" ]; then
+    echo "Unrecognized target [$PACKAGE_OS.$PACKAGE_ARCH]" >&2
+    exit 1
+fi
+
+case "$PACKAGE_ARCH" in
+    'amd64')
+        RUST_TARGET='x86_64-unknown-linux-musl'
+        ;;
+
+    'arm32v7')
+        RUST_TARGET='armv7-unknown-linux-gnueabihf'
+        ;;
+
+    'aarch64')
+        RUST_TARGET='aarch64-unknown-linux-gnu'
+        ;;
+esac
+
+if [ -n "$RUST_TARGET" ]; then
+    RUST_TARGET_COMMAND="rustup target add $RUST_TARGET &&"
+fi
+
+
+case "$PACKAGE_OS.$PACKAGE_ARCH" in
+    alpine.amd64)
+        # TODO: How do we handle these important dependencies so that they get updated?
+        # The below SETUP was copied from https://github.com/emk/rust-musl-builder/blob/master/Dockerfile.
+        # The beginning of the below setup specifies versions for other dependencies.
+        # Here are the places to check for new releases:
+        #
+        # - https://github.com/rust-lang/mdBook/releases
+        # - https://github.com/EmbarkStudios/cargo-about/releases
+        # - https://github.com/EmbarkStudios/cargo-deny/releases
+        SETUP_COMMAND=$'
+            OPENSSL_VERSION=1.1.1g
+            MDBOOK_VERSION=0.3.7
+            CARGO_ABOUT_VERSION=0.2.2
+            CARGO_DENY_VERSION=0.6.7
```

Comment:
We need to find a way to handle these important dependencies since they will get updated. Any ideas? We can elevate to the top of file for clarity, but that will separate it from the alpine logic.